### PR TITLE
Make duration field optional and set required None for existing data

### LINKED ISSUE #2274: Stop making "project duration" field required by Wagtail system
Right now the Wagtail system enforces the "project duration" field to be incuded (along with application title, applicant name, and applicant email address).

If a user tried to create a form, and does not add this field, their work cannot be saved.

It is possible that the way the project duration question is asked in the future will need to be changed, so we're making this field no longer a Wagatil system enforced required field.

### PR DESCRIPTION
**Pull request by sandeepsajan0**
Fixes #2274 and Closes #2337 

Add a custom migration to set required False for existing data, otherwise, it was saved None by default for existing data.

### CODE REVIEW

**Comment by frjo** [comment]
This PR has migrations that conflicts with https://github.com/HyphaApp/hypha/pull/2678 so we wait a bit with this one.

**Comment by sandeepsajan0** [comment]
@frjo I have merged the migrations. Looks like #2678 is stable now, so we may proceed with this as well?

**Comment by frjo** [comment]
Excellent, I have deployed it to test now.

**Comment by frjo** [comment]
Deployed failed, I have rebuilt the migration now.

**Comment by frjo** [comment]
Missed the custom migration you built, readded now.

File: hypha/apply/funds/migrations/0093_set_duration_false.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from django.db import migrations
+
+
+def set_duration_required_false(apps, schema_editor):
+    ApplicationSubmission = apps.get_model('funds', 'ApplicationSubmission')
+    for submission in ApplicationSubmission.objects.all():
+        for id, struct_child in enumerate(submission.form_fields):
+            struct_value = struct_child.value
+            if struct_value['field_label'] == 'Duration' and struct_value['required'] is None:
```
Comment (frjo) [inline hypha/apply/funds/migrations/0093_set_duration_false.py]: I believe this is looking for fields that has the label set to "Duration". On the test site I get this migration error.

`TypeError: 'RichText' object is not subscriptable`

I assume there exist other kinds of fields on some applications that is named "Duration" but is to a duration field.

Can we make the migration only look for explicit Duration fields?

**Review by frjo — CHANGES_REQUESTED** [review]

File: hypha/apply/funds/migrations/0093_set_duration_false.py (hunk 1 above)
Comment (sandeepsajan0) [reply hypha/apply/funds/migrations/0093_set_duration_false.py]: @frjo I have changed that lookup from `struct_value['field_label']` to `struct_child.block_type`. I think it should work fine now. Can we test it once again?

**Comment by frjo** [comment]
Perfect! Deploying to test now.

**Review by frjo — APPROVED** [review]

**Pull request merged**